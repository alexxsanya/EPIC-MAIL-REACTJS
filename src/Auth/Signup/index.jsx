import React from 'react';
import {
  Form,
  Row,
  Col,
  Button,
  InputGroup,
} from 'react-bootstrap';
import { connect } from 'react-redux';
import Loader from 'react-loader';
import '../CSS/Auth.scss';
import signupUser from '../Redux/actions/signupAction';
import StatusLabel from '../../common/components/StatusLabel';

class Signup extends React.Component {
  constructor (props){
    super(props);
    this.state = {
      email: '',
      username: '',
      firstname: '',
      lastname: '',
      password: '',
      isSubmitted: true,
    }
  }

  handleChange = event => {
    const {name, value} = event.target;
    this.setState({
      [name]: value,
    });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    this.setState({
        isSubmitted: false
      })
    const { username, password, firstname, lastname,} = this.state
    const userData = {
        username: username,
        password: password,
        email: `${username}@epicmail.com`,
        firstname: firstname,
        lastname: lastname,
        recovery_email: `${username}_${firstname}@gmail.com`,
    }

    this.props.signupUser(userData);
  }

  render() {
    const { password, username,firstname, lastname } = this.state;
    const { errorOccured, userInfo, response, isSigningUp } = this.props;

    if (userInfo.status === 201) {
      localStorage.setItem("user_token", userInfo.data[0].token);
      localStorage.setItem("username", firstname);
      setTimeout(() => {
        window.location = `/dashboard`;
      },
      300000);
    }

    return (
      <div className="signup-component">
        <h2 className="section-title">Create Account</h2>
        <Row className="status-section">
          <Col sm={12}>
              {
                errorOccured.errorMessage === undefined
                && (
                <StatusLabel
                message={String(response)}
                tip="Check Your Network & Try Again"
                />
                )
              }
              {
                (userInfo.data === [] || userInfo.status === 400)
                && (
                <StatusLabel
                message={userInfo.error}
                tip="try again"
                />
                )
              }
              {
                (userInfo.status === 201)
                && <StatusLabel 
                message="Account successfully created" 
                tip=" "
                variant="success"
                />
              }
          </Col>
        </Row>
        <Form onSubmit={this.handleSubmit}>
        <Form.Group controlId="formBasicFirstname">
            <Form.Control
              type="text"
              required
              name="firstname"
              placeholder="Enter firstname"
              value={firstname}
              onChange={this.handleChange}
            />
          </Form.Group>        
        <Form.Group controlId="formBasicLastname">
            <Form.Control
              type="text"
              required
              name="lastname"
              placeholder="Enter lastname"
              value={lastname}
              onChange={this.handleChange}
            />
          </Form.Group>
          <InputGroup.Prepend className="username_section">
            <Form.Control
              type="text"
              required
              name="username"
              placeholder="Enter username"
              value={username}
              onChange={this.handleChange}
              className="username_input"
            />
            <InputGroup.Text id="inputGroupPrepend">@epicmail.com</InputGroup.Text>
          </InputGroup.Prepend>
          <Form.Group controlId="formBasicTitle">
            <Form.Control
              type="password"
              required
              name="password"
              placeholder="Enter password"
              value={password}
              onChange={this.handleChange}
            />
          </Form.Group>
          <Row>
            <Col sm={12} className="call-for-action">
              <Loader loaded={!isSigningUp} color="#3379f5">
                <Button type="submit" id="submitForm">
                    Create Account
                </Button>
              </Loader>
            </Col>
          </Row>
          <Row>
            <Col sm={12} className="label-action">
              <span> You have an account, click <a href='/login'>here</a> to login</span>
            </Col>
          </Row>
        </Form>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
    isSigningUp: state.signupReducer.isSigningUp,
    errorOccured: state.signupReducer.error,
    isSignedUp: state.signupReducer.status,
    userInfo: state.signupReducer.item,
    response: state.signupReducer.message
  });
  
export default connect(mapStateToProps, { signupUser })(Signup);