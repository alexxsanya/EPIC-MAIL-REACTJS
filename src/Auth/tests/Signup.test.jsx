import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Signup } from '../Signup';

const props = {
  isSigningUp: false,
  errorOccured: {
    errorMessage: 'No Message',
  },
  isSignedUp: false,
  userInfo: {},
  response: 'Signup Failed',
};

describe('Signup Component', () => {
  it('should match the snapshot', () => {
    const component = shallow(<Signup {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should call handleSubmit upon Signup ', () => {
    const event = {
      preventDefault: jest.fn(),
    };
    const handleSubmitMock = jest.fn(event);
    const wrapper = mount(<Signup {...props} />);

    expect(wrapper).toBeDefined();
    // expect(handleSubmitMock).toBeCalled();
  });
});
