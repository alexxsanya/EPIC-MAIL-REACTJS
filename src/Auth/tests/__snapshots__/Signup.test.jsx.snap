// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Component should match the snapshot 1`] = `
<div
  className="signup-component"
>
  <h2
    className="section-title"
  >
    Create Account
  </h2>
  <ForwardRef
    className="status-section"
    noGutters={false}
  >
    <Col
      sm={12}
    />
  </ForwardRef>
  <Form
    inline={false}
    onSubmit={[Function]}
  >
    <FormGroup
      controlId="formBasicFirstname"
    >
      <FormControl
        name="firstname"
        onChange={[Function]}
        placeholder="Enter firstname"
        required={true}
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      controlId="formBasicLastname"
    >
      <FormControl
        name="lastname"
        onChange={[Function]}
        placeholder="Enter lastname"
        required={true}
        type="text"
        value=""
      />
    </FormGroup>
    <InputGroupPrepend
      className="username_section"
    >
      <FormControl
        className="username_input"
        name="username"
        onChange={[Function]}
        placeholder="Enter username"
        required={true}
        type="text"
        value=""
      />
      <InputGroupText
        id="inputGroupPrepend"
      >
        @epicmail.com
      </InputGroupText>
    </InputGroupPrepend>
    <FormGroup
      controlId="formBasicTitle"
    >
      <FormControl
        name="password"
        onChange={[Function]}
        placeholder="Enter password"
        required={true}
        type="password"
        value=""
      />
    </FormGroup>
    <ForwardRef
      noGutters={false}
    >
      <Col
        className="call-for-action"
        sm={12}
      >
        <Component
          color="#3379f5"
          component="div"
          loaded={true}
          loadedClassName="loadedContent"
          parentClassName="loader"
        >
          <Button
            active={false}
            disabled={false}
            id="submitForm"
            type="submit"
            variant="primary"
          >
            Create Account
          </Button>
        </Component>
      </Col>
    </ForwardRef>
    <ForwardRef
      noGutters={false}
    >
      <Col
        className="label-action"
        sm={12}
      >
        <span>
           You have an account, click 
          <a
            href="/login"
          >
            here
          </a>
           to login
        </span>
      </Col>
    </ForwardRef>
  </Form>
</div>
`;
